ftw/publisher/core/tests/test_portlet_adapter.py

```diff
@@ -208,3 +208,27 @@ def test_portlets_with_settings(self):
             IPortletAssignmentSettings.providedBy(settings),
             'Portlet settings is not PortletAssignmentSettings but %s' % (
                 type(settings)))
+
+    def test_portlet_visibility_settings(self):
+        assignment = self.layer['right_portlets'].get('title2', False)
+        settings = {'visible': True}
```

Comment:
Did you verify that in "production" this is actually a dict?
I would have expected it to be a `PersistentMapping`.
If it is a persistent data structure could you change it here too in order to detect json problems?